### ISSUE TITLE
GError [TypeError]: Cannot read properties of undefined (reading 'Unknown')

### ISSUE THREAD

**Issue by phoenixgao** [wechaty/puppet-xp#43]
Getting this error while running a 6 line bot and pointed to
https://github.com/wechaty/puppet-xp/blob/main/src/puppet-xp.ts#L50

It seems
`wechaty-puppet` not longer export `ContactType`, `ContactGender` and `MessageType` etc, they are all wrapped in a `type` property, see

https://github.com/wechaty/puppet/blob/main/src/mods/mod.ts#L46
and 
https://github.com/wechaty/puppet/blob/main/src/mods/types.ts#L26

Should use something like 
```js
import { type } from 'wechaty-puppet'
const gender = type.ContactGender.Unknown
```
here?

And

```js
import { FileBoxType } from 'file-box'
```

**Comment by huan** [comment]
I guess you are using `wechaty@1.x` with `wechaty-puppet-xp@0.x`.

Please downgrade your Wechaty version from 1.x to 0.x and see if your problem can be solved.

We have breaking changes in the v1.x and I hope the XP can upgrade to be compatible with the Wechaty Puppet v1.0 API soon.

**Comment by phoenixgao** [comment]
@huan 
Yes was using `wechaty@1.x` as it got installed by default running `npm i`.
However downgrading my wechaty to 0.79 gets new issues when trying the ripe-wechaty example,
First it's the ScanStatus is not exported in wechaty@0.x, so I changed to correctly import from wechaty-puppet,
but then I got bunch of type not match errors like:

```shell
TSError: ⨯ Unable to compile TypeScript:
xp.ts:76:3 - error TS2322: Type 'PuppetXp' is not assignable to type '"wechaty-puppet-puppeteer" | "wechaty-puppet-service" | "wechaty-puppet-mock" | "wechaty-puppet-wechat" | "wechaty-puppet-wechat4u" | "wechaty-puppet-padlocal" | "wechaty-puppet-xp" | ... 9 more ... | undefined'.
  Type 'PuppetXp' is not assignable to type 'PuppetInterface'.
    Types of property 'roomMemberSearch' are incompatible.
      Type '((roomId: string, query: string | RoomMemberQueryFilter | unique symbol) => Promise<string[]>) & ((roomId: string, query: string | RoomMemberQueryFilter | unique symbol) => Promise<...>)' is not assignable to type '((roomId: string, query: string | unique symbol | RoomMemberQueryFilter) => Promise<string[]>) & ((roomId: string, query: string | unique symbol | RoomMemberQueryFilter) => Promise<...>)'.
        Type '((roomId: string, query: string | RoomMemberQueryFilter | unique symbol) => Promise<string[]>) & ((roomId: string, query: string | RoomMemberQueryFilter | unique symbol) => Promise<...>)' is not assignable to type '(roomId: string, query: string | unique symbol | RoomMemberQueryFilter) => Promise<string[]>'.
          Types of parameters 'query' and 'query' are incompatible.
            Type 'string | unique symbol | RoomMemberQueryFilter' is not assignable to type 'string | RoomMemberQueryFilter | unique symbol'.
              Type 'unique symbol' is not assignable to type 'string | RoomMemberQueryFilter | unique symbol'.

```

As a new wechaty user I found dependencies' version requirement is a bit hard to track, if a working set of fixed version numbers could be added to the readme would be very helpful.

**Comment by huan** [comment]
Thanks for the reporting and the suggestion.

I think we should have a v0.x branch for our getting-started repo so that you can just clone & run.

Will do it tonight for you.

**Comment by huan** [comment]
We have a wechaty@0.x getting started repo branch created at https://github.com/wechaty/getting-started/tree/v0.x .

If anyone need to use wechaty@0.x before wechaty@1.x stable, please try that branch first, and it should be able to work out-of-the-box.

Create issue to report any problem if it does not work for you, thank you very much.

**Comment by phoenixgao** [comment]
0.68 doesn't work for me, and the getting started branch can't use the latest puppet-xp either.
I managed to make my local code works with wechaty 1.x by fixing all the imports, since this repo is under heavy development so I'll close this issue for now.

**Comment by huan** [comment]
It's great to know that you have fixed this repo to compatible with v1.0, congratulations!

May I ask you to send your fixes back as a PR to this repo so that we can make XP compatible with the latest version?
